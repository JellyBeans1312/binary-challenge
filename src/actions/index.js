export const addEvents = (events) => ({
  type: 'ADD_EVENTS',
  events
});