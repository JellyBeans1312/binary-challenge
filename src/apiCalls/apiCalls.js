export const getEvents = async () => {
  try {
    const options = { 
      method: 'GET',
      headers: {
        Authorization: 'Bearer qJhnqW5Rsc36m6cUYXZyCaV2BaFNB0QoGpr4fmk_',
        Accept: 'application/json',
        'Content-Type': 'application/json'
      }
    }

    const response = await fetch('https://api.predicthq.com/v1/events/', options)
    const result = await response.json()
    return result.results
  } catch(error) {
    console.log(error);
  }
}

export const searchEvent = async (searchParam) => {
  try {
    const options = {
      method: 'GET',
      headers: {
        Authorization: 'Bearer qJhnqW5Rsc36m6cUYXZyCaV2BaFNB0QoGpr4fmk_',
        Accept: 'application/json',
        'Content-Type': 'application/json'
      }
    }
    
    const response = await fetch(`https://api.predicthq.com/v1/events/?q=${searchParam}/`, options)
    const result = await response.json()
    return result.results

  } catch(error) {
    throw new Error(error)
  }
}

export const getAddress = async (lat, long) => {
  try {
    const response = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${long}&key=AIzaSyDPPlNyjATG8sQnfsdg45ln7kUoyFH24m0`)
    const result = await response.json();
    return result.results.map(result => {
      return result.formatted_address
    });
  } catch(error) {
    console.log(error)
  }
}