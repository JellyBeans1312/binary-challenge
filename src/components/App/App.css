@import url('https://fonts.googleapis.com/css?family=Roboto+Condensed:400,700&display=swap');

.App {
  text-align: center;
  font-family: 'Roboto Condensed', sans-serif;
  font-weight: 700;
}

.App-header {
  color: #fff;
  font-size: 3rem;
  background-image: url('../../images/navbardesign.svg');
  background-size: cover;
  width: 100%;
  height: 40vh;
}

.App-link {
  color: #61dafb;
}

