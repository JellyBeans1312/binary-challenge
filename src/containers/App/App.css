@import url('https://fonts.googleapis.com/css?family=Roboto+Condensed:400,700&display=swap');

.App {
  text-align: center;
  font-family: 'Roboto Condensed', sans-serif;
  font-weight: 700;
}

.App-header {
  color: #fff;
  font-size: 3rem;
  background-image: url('../../images/navbardesign.svg');
  background-size: cover;
  width: 100%;
  height: 40vh;
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-template-rows: .1fr;
  grid-column-gap: 0px;
  grid-row-gap: 0px;
}

.App-link {
  color: #61dafb;
}

h1 {
  padding: 0px;
  margin: 0px;
  display: inline-block
}
.app-name {
  grid-area: 1 / 2 / 2 / 5;
  margin: auto;
}

.app-component {
  grid-area: 1 / 5 / 2 / 6;
  margin: auto;
}

form input{
  height: 30px;
  width: 220px;
}

form button {
  height: 37px;
  color: white;
  background-color: #5C10B9;
  border-color: white;
}