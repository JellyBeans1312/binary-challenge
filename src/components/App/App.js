import React, { Component } from 'react';
import './App.css';
import { getEvents } from './apiCalls/apiCalls';
import { Navbar } from './Navbar/Navbar'

class App extends Component {

  async componentDidMount() {
    const events = await getEvents()
    console.log(events)
  }
  render() {
    return (

    <div className="App">
      <header className="App-header">
      <Navbar />
      </header>
    </div>
  );
  }
}

export default App;
