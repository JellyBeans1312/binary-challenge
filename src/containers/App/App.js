import React, { Component } from 'react';
import './App.css';
import { getEvents } from '../../apiCalls/apiCalls';
import { connect } from 'react-redux';
import { addEvents } from '../../actions';
import Search from '..//Search/Search';
import { Navbar } from '../../components/Navbar/Navbar'
import PropTypes from 'prop-types';

export class App extends Component {

  async componentDidMount() {
    const events = await getEvents()
    this.props.addEvents(events)
  }

  render() {
    return (

    <div className="App">
      <header className="App-header">
        <h1 className='app-name'>Event Finder</h1>
        <span className='app-component'>
          <Search/>
        </span>
      </header>
      <Navbar addEvents={this.props.addEvents}/>
    </div>
  );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addEvents: events => dispatch(addEvents(events))
});

export default connect(null, mapDispatchToProps)(App);


App.propTypes = {
  addEvents: PropTypes.func.isRequired
}