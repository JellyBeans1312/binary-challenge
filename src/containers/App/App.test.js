import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps } from './App';
import { getEvents } from '../../apiCalls/apiCalls'
import { addEvents } from '../../actions';

jest.mock('../../apiCalls/apiCalls')

describe('App', () => {
  let wrapper;
  const mockEvents = [
    { title: "something", category: 'concert', location: [{lat: 23, long: 234}] }
  ];

  beforeEach(() => {
     wrapper = shallow(
      <App
        events={mockEvents}
      />
    )

    getEvents.mockImplementation(() => {
      return Promise.resolve({mockEvents})
    });
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot() 
  });

  it('should call getEvents', async () => {
    expect(getEvents).toHaveBeenCalled()
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with addEvents action', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = addEvents(mockEvents)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.addEvents(mockEvents)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    });
  })
});

