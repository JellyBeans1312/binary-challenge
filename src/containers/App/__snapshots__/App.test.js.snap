// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="app-name"
    >
      something
    </h1>
    <span
      className="app-component"
    >
      <Connect(Search) />
    </span>
  </header>
  <Navbar />
</div>
`;
